# Churn by Customer Segment – Management Analysis

Notebook ini bertujuan mengidentifikasi segmen customer dengan tingkat churn yang secara signifikan lebih tinggi dibanding baseline keseluruhan. 
Hasil analisis digunakan untuk membantu manajemen menentukan prioritas segmen yang paling layak menjadi fokus strategi retensi.


In [1]:
import pandas as pd

url = "https://raw.githubusercontent.com/AlvitoDwiP/churn-analysis/main/churn/data/raw/WA_Fn-UseC_-Telco-Customer-Churn.csv"
df = pd.read_csv(url)


## Overall Churn Baseline

Tingkat churn keseluruhan digunakan sebagai baseline untuk menilai apakah suatu segmen memiliki risiko churn yang relatif lebih tinggi atau lebih rendah dibanding rata-rata populasi customer.


In [2]:
baseline_churn = df['Churn'].value_counts(normalize=True)['Yes']
baseline_churn


np.float64(0.2653698707936959)

Baseline churn sebesar 26.5% menjadi titik acuan untuk mengevaluasi tingkat risiko churn pada masing-masing segmen customer.


## Churn by Contract Type
Analisis ini membandingkan tingkat churn berdasarkan jenis kontrak untuk mengidentifikasi perbedaan risiko churn antar tingkat komitmen berlangganan.


In [3]:
churn_contract = (
    df.groupby('Contract')['Churn']
      .value_counts(normalize=True)
      .unstack()
      .assign(risk_index=lambda x: x['Yes'] / baseline_churn)
      .sort_values('Yes', ascending=False)
)

churn_contract


Churn                 No       Yes  risk_index
Contract                                      
Month-to-month  0.572903  0.427097    1.609440
One year        0.887305  0.112695    0.424672
Two year        0.971681  0.028319    0.106714

Customer dengan kontrak month-to-month memiliki churn 43% atau sekitar 1.6 kali lebih tinggi dibanding baseline. 
Sebaliknya, kontrak jangka panjang menunjukkan churn yang jauh lebih rendah menandakan efek protektif dari komitmen berlangganan.


## Churn by Internet Service
Analisis ini bertujuan untuk melihat perbedaan tingkat churn berdasarkan jenis layanan internet yang digunakan customer.


In [4]:
churn_internet = (
    df.groupby('InternetService')['Churn']
      .value_counts(normalize=True)
      .unstack()
      .assign(risk_index=lambda x: x['Yes'] / baseline_churn)
      .sort_values('Yes', ascending=False)
)

churn_internet


Churn                  No       Yes  risk_index
InternetService                                
Fiber optic      0.581072  0.418928    1.578656
DSL              0.810409  0.189591    0.714441
No               0.925950  0.074050    0.279044

Customer pengguna layanan fiber optic menunjukkan churn tertinggi 42% atau sekitar 1.6 kali baseline.
Hal ini mengindikasikan bahwa layanan dengan harga atau ekspektasi lebih tinggi berpotensi memiliki risiko churn yang lebih besar.


## Churn by Senior Citizen
Variabel SeniorCitizen bernilai 1 untuk customer senior dan 0 untuk non-senior. Analisis ini membandingkan tingkat churn antara kedua kelompok tersebut.


In [5]:
churn_senior = (
    df.groupby('SeniorCitizen')['Churn']
      .value_counts(normalize=True)
      .unstack()
      .assign(risk_index=lambda x: x['Yes'] / baseline_churn)
)

churn_senior


Churn                No       Yes  risk_index
SeniorCitizen                                
0              0.763938  0.236062    0.889557
1              0.583187  0.416813    1.570686

Customer senior memiliki tingkat churn hampir dua kali lipat dibanding non-senior.
Temuan ini menunjukkan bahwa faktor usia atau kebutuhan khusus customer dapat memengaruhi retensi.


## Segment Prioritization for Retention Strategy

Berdasarkan perbandingan terhadap baseline churn, segmen dengan prioritas tertinggi untuk strategi retensi adalah:
1. Customer dengan kontrak month-to-month
2. Customer pengguna layanan fiber optic
3. Customer senior citizen

Segmen-segmen ini menunjukkan tingkat churn yang secara konsisten lebih tinggi dibanding rata-rata, sehingga memberikan potensi dampak terbesar jika dilakukan intervensi retensi.
